docs/topics/downloader-middleware.rst

```diff
@@ -429,6 +429,40 @@ In order to use this policy, set:
 * :setting:`HTTPCACHE_POLICY` to ``scrapy.extensions.httpcache.RFC2616Policy``
 
 
+.. _httpcache-policy-custom:
+
+Writing your own policy
+~~~~~~~~~~~~~~~~~~~~~~~
+
+You can implement a cache policy by creating a class that
+defines the methods described below.
+
+.. module:: scrapy.extensions.httpcache
+
+.. class:: CachePolicy
+
+    .. method:: should_cache_request(request)
+
+      Return True if the :class:`~scrapy.http.Request` is cacheable otherwise False
+
+    .. method:: should_cache_response(response, request)
+
+      Return True if :class:`~scrapy.http.Response` should be cached otherwise False
+
+    .. method:: is_cached_response_fresh(cachedresponse, request)
+
+      Return True if cached :class:`~scrapy.http.Response` has expired otherwise False
+
+    .. method:: is_cached_response_valid(cachedresponse, response, request)
+
+      Return True if cached :class:`~scrapy.http.Response` is valid otherwise False
```

Comment:
It is my understanding that `is_cached_response_valid` receives a new response and the previously-cached version of that same response, and you must return `True` if the cached response should be used instead of the fresh response, or `False` if the new response should be used and added to the cache, overriding the previously-cached response.

I think the description should make this clear.